spec/acceptance/class_spec.rb

```diff
@@ -41,19 +16,37 @@ class { 'apache':
         default_vhost => false,
         purge_configs => true,
       }
-      if $facts['os']['family'] == 'RedHat' {
-        include epel
-        class { 'apache::mod::wsgi': wsgi_socket_prefix => '/var/run/wsgi' }
-      } else {
-        class { 'apache::mod::wsgi': }
+      if $facts['os']['release']['major'] == '20.04' {
+        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.8",}
+      } elsif $facts['os']['release']['major'] == '18.04' {
+        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.6",}
+      } elsif $facts['os']['release']['major'] == '16.04' {
+        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.5",}
+      } elsif $facts['os']['release']['major'] == '10' {
+        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.7",}
+      } elsif $facts['os']['release']['major'] == '9' {
+        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.5",}
+      }
```

Comment:
```suggestion
      if $facts['os']['family'] == 'Debian' {
        $mod_path = $facts['os']['release']['major'] ? {
          '18.04' => "/usr/lib/apache2/modules/mod_wsgi.so-3.6",
          '16.04' => "/usr/lib/apache2/modules/mod_wsgi.so-3.5",
          '10'    => "/usr/lib/apache2/modules/mod_wsgi.so-3.7",
          '9'     => "/usr/lib/apache2/modules/mod_wsgi.so-3.5",
          default => undef,
        }
        $wsgi = {
          'package_name' => "libapache2-mod-wsgi-py3",
          'mod_path'     => $mod_path,
        }
      } else {
        $wsgi = {}
      }
```

Also wondering if you can reuse `$puppetboard::python_version` here for `$mod_path`

Reply:
good catch. I'm currently testing this locally:

```diff
diff --git a/spec/acceptance/class_spec.rb b/spec/acceptance/class_spec.rb
index a2fd425..4623e6c 100644
--- a/spec/acceptance/class_spec.rb
+++ b/spec/acceptance/class_spec.rb
@@ -16,16 +16,9 @@ describe 'puppetboard class' do
         default_vhost => false,
         purge_configs => true,
       }
-      if $facts['os']['release']['major'] == '20.04' {
-        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.8",}
-      } elsif $facts['os']['release']['major'] == '18.04' {
-        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.6",}
-      } elsif $facts['os']['release']['major'] == '16.04' {
-        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.5",}
-      } elsif $facts['os']['release']['major'] == '10' {
-        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.7",}
-      } elsif $facts['os']['release']['major'] == '9' {
-        $wsgi = {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-3.5",}
+      $wsgi = $facts['os']['family'] ? {
+        'Debian' => {package_name => "libapache2-mod-wsgi-py3", mod_path => "/usr/lib/apache2/modules/mod_wsgi.so-${puppetboard::python_version}"},
+        default  => {},
       }
       class { 'apache::mod::wsgi':
         * => $wsgi,

```

Reply:
I updated this to rely on the nonversioned .so file. We canot easily access puppetboard::python_version. The .so file could be provided by other packages as well, but they always conflict with libapache2-mod-wsgi-py3. If libapache2-mod-wsgi-py3 got installed we know that the .so file is the correct one.